CodeEdit/Features/Editor/Models/EditorManager.swift

```diff
@@ -36,6 +36,17 @@ class EditorManager: ObservableObject {
     var tabBarTabIdSubject = PassthroughSubject<String?, Never>()
     var cancellable: AnyCancellable?
 
+    // This caching mechanism is a temporary solution and is not optimized
+    var updateCachedFlattenedEditors: Bool = true
+    var cachedFlettenedEditors: [Editor] = []
+    var flattenedEditors: [Editor] {
+        if updateCachedFlattenedEditors {
+            cachedFlettenedEditors = self.getFlattened()
+            updateCachedFlattenedEditors = false
+        }
+        return cachedFlettenedEditors
+    }
```

Comment:
Forgot to set `updateCachedFlattenedEditors` back to true whenever we update the editors

Reply:
That would explain the stale data!